src/compiler/transformers/esDecorators.ts

```diff
@@ -2332,4 +2340,59 @@ export function transformESDecorators(context: TransformationContext): (x: Sourc
             factory.createStringLiteralFromNode(name);
     }
+
+    function createMetadata(name: Identifier, hasProto: boolean) {
+        const varDecl = factory.createVariableDeclaration(
+            name,
+            /*exclamationToken*/ undefined,
+            /*type*/ undefined,
+            factory.createConditionalExpression(
+                factory.createBinaryExpression(
+                    factory.createBinaryExpression(factory.createTypeOfExpression(factory.createIdentifier("Symbol")), factory.createToken(SyntaxKind.EqualsEqualsEqualsToken), factory.createStringLiteral("function")),
+                    factory.createToken(SyntaxKind.AmpersandAmpersandToken),
+                    factory.createPropertyAccessExpression(factory.createIdentifier("Symbol"), factory.createIdentifier("metadata")),
+                ),
+                factory.createToken(SyntaxKind.QuestionToken),
+                factory.createCallExpression(
+                    factory.createPropertyAccessExpression(factory.createIdentifier("Object"), factory.createIdentifier("create")),
+                    /*typeArguments*/ undefined,
+                    [hasProto ? createSymbolMetadataReference() : factory.createNull()]
+                ),
+                factory.createToken(SyntaxKind.ColonToken),
+                factory.createIdentifier("undefined"),
+            ),
+        );
+        return factory.createVariableStatement(/*modifiers*/ undefined, factory.createVariableDeclarationList([varDecl], NodeFlags.Const));
+    }
+
+    function createSymbolMetadata(target: Identifier | ThisExpression, value: Identifier) {
+        const properties = [
+            factory.createPropertyAssignment("configurable", factory.createTrue()),
+            factory.createPropertyAssignment("writable", factory.createTrue()),
+            factory.createPropertyAssignment("enumerable", factory.createTrue()),
+            factory.createPropertyAssignment("value", value),
+        ];
+
+        const defineProperty = factory.createCallExpression(
+            factory.createPropertyAccessExpression(factory.createIdentifier("Object"), "defineProperty"),
+            /*typeArguments*/ undefined,
+            [
+                target,
+                factory.createPropertyAccessExpression(factory.createIdentifier("Symbol"), factory.createIdentifier("metadata")),
+                factory.createObjectLiteralExpression(properties, /*multiLine*/ false),
+            ]
+        );
+
+        return setEmitFlags(
+            factory.createIfStatement(value, factory.createExpressionStatement(defineProperty)),
+            EmitFlags.SingleLine,
+        );
+    }
+
+    function createSymbolMetadataReference() {
+        return factory.createElementAccessExpression(
+            factory.createThis(),
+            factory.createPropertyAccessExpression(factory.createIdentifier("Symbol"), factory.createIdentifier("metadata")),
+        );

```

Comment:
Per the [spec text](https://github.com/pzuraq/ecma262/pull/10/files#diff-181371b08d71216599b0acccbaabd03c306da6de142ea6275c2135810999805aR25266-R25268), parent metadata should be resolved against the super constructor, not against `this`.